### PR TITLE
Increasing test coverage

### PR DESCRIPTION
**Pull request by daniloaf**

### CODE REVIEW

File: .coveragerc [hunk 1]
```diff
@@ -1,6 +1,11 @@
 [run]
 source = openapi
-omit = openapi/reload.py
+omit =
+    openapi/reload.py
+    openapi/db/migrations.py
```
Comment (lsbardel) [inline .coveragerc]: why are you omitting these below?

File: openapi/data/fields.py [hunk 2]
```diff
@@ -100,7 +100,7 @@ def date_time_field(required=False):
 # VALIDATORS
 
 
-def email_validator(field, value, data):
```
Comment (lsbardel) [inline openapi/data/fields.py]: why are you removing the data? it may be required for more complex validators

File: openapi/data/fields.py [hunk 3]
```diff
@@ -222,9 +222,21 @@ def dump(self, value):
...
+class BoolValidator(Validator):
+
+    def __call__(self, field, value):
+        value = str(value).lower()
+        if value not in ('true', 'false'):
+            raise ValidationError(field.name, '%s not valid' % value)
+        return value == 'true'
+
+    def dump(self, value):
+        return str(value).lower()
```
Comment (lsbardel) [inline openapi/data/fields.py]: I don't think this is right.
return ``bool(value)`` makes more sense to me

File: .coveragerc (hunk 1 above)
Comment (daniloaf) [reply .coveragerc]: They seem more operational code. I don't think it's worth to add tests for those

File: openapi/data/fields.py (hunk 2 above)
Comment (daniloaf) [reply openapi/data/fields.py]: There were no validators using it, so I removed. We can add it if we need in the future

File: openapi/data/fields.py (hunk 3 above)
Comment (daniloaf) [reply openapi/data/fields.py]: the dump should result in a json valid format, no? In this case, `"true"` or `"false"`

Comment (lsbardel) [reply openapi/data/fields.py]: a boolean is a valid JSON value.
It needs to be `True` or `False`

File: openapi/data/fields.py [hunk 4]
```diff
@@ -110,7 +110,7 @@ def email_validator(field, value, data):
 class Validator:
     dump = None
 
-    def __call__(self, field, value, data):
+    def __call__(self, field, value):
         raise ValidationError(field.name, 'inavlid')
```
Comment (jklepatch) [inline openapi/data/fields.py]: inavlid -> invalid

File: openapi/data/fields.py [hunk 5]
```diff
@@ -222,9 +222,21 @@ def dump(self, value):
...
+        return super().__call__(field, value)
+
+
+class BoolValidator(Validator):
+
+    def __call__(self, field, value):
+        value = str(value).lower()
+        if value not in ('true', 'false'):
+            raise ValidationError(field.name, '%s not valid' % value)
+        return value == 'true'
```
Comment (jklepatch) [inline openapi/data/fields.py]: For other validators we return the value, but for this one we do a comparison. Is this what we want?

File: openapi/data/fields.py (hunk 3 above)
Comment (jklepatch) [reply openapi/data/fields.py]: I dont think that `True` or `False` are valid json values.

``` 
A JSON value MUST be an object, array, number, or string, or one of
   the following three literal names:

      false null true
```

https://tools.ietf.org/html/rfc7159

File: tests/data/test_fields.py [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+    value = uuid4()
+    validator = UUIDValidator()
+    assert validator.dump(value) == value.hex
+    assert validator.dump(value.hex) == value.hex
+
+
+def test_EnumValidator_call_valid():
+    field = enum_field(FakeEnum)
+    validator = EnumValidator(FakeEnum)
+    assert validator(field, None) is None
```
Comment (jklepatch) [inline tests/data/test_fields.py]: Not directly related to the tests, but I am not sure why we would not raise in this case

File: openapi/data/fields.py (hunk 3 above)
Comment (lsbardel) [reply openapi/data/fields.py]: @jklepatch they don't need to be Json type, they need to be types python know how to convert to JSON.
@daniloaf, please return a boolean. It is the correct return type.
These values are part of the dictionary which is subsequently serialised via JSON.
```python
json.dumps({'whatever': True})
'{"whatever": true}'
```

Comment (lsbardel) [reply openapi/data/fields.py]: tests should be able to help here too. Try seruializing after `dump`

File: openapi/data/fields.py (hunk 5 above)
Comment (daniloaf) [reply openapi/data/fields.py]: The comparison will return `True` or `False`

File: openapi/data/fields.py (hunk 3 above)
Comment (daniloaf) [reply openapi/data/fields.py]: So I believe some validators are wrong, like DateTimeValidator as it returns the datetime isoformat. Should I change those?

File: tests/data/test_fields.py (hunk 6 above)
Comment (daniloaf) [reply tests/data/test_fields.py]: I had to allow `None` on this validator before for some reason, but I don't know if it's necessary anymore. I'll check if we can raise on this case

**Pull request merged**